src/test/java/com/marklogic/client/ext/file/CascadeCollectionsAndPermissionsFileProcessorTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.client.ext.file;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.ext.AbstractIntegrationTest;
+import com.marklogic.client.io.DocumentMetadataHandle;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CascadeCollectionsAndPermissionsFileProcessorTest extends AbstractIntegrationTest {
+
+	final private String PARENT_COLLECTION = "ParentCollection";
+	final private String CHILD_COLLECTION = "ChildCollection";
+
+	@BeforeEach
+	public void setup() {
+		client = newClient(MODULES_DATABASE);
+		DatabaseClient modulesClient = client;
+		modulesClient.newServerEval().xquery("cts:uris((), (), cts:true-query()) ! xdmp:document-delete(.)").eval();
+	}
+
+	@Test
+	public void parentSettingsOnlyCascadeWhenChildHasNoSettings() {
+		String directory = "src/test/resources/process-files/cascading-metadata-test/parent1-withCP";
+		GenericFileLoader loader = new GenericFileLoader(client);
+		List<DocumentFile> files = loader.loadFiles(directory);
+
+		Optional<DocumentFile> optionalTestFile = files.stream().filter(file -> "/child1_1-noCP/test.json".equals(file.getUri())).findFirst();
```

Comment:
It's okay to test the DocumentFile objects that are returned, but what we really care about is that the modules were loaded correctly - users care about that, ml-gradle users will never see the List. Use the `verifyCollections` and `verifyPermissions` methods I added - they'll make the test much shorter and easier to read. If you want, include assertions on the List that's returned as well, though we can safely assume that if the modules were loaded correctly, then the List is going to be correct as well (I would not bother with those assertions).